### PR TITLE
Fixes #10905: Zip promises post generation for dsc agent

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/10905

### CODE REVIEW

File: rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+if [ -z "${RUDDER_NEXT_POLICIES_DIRECTORY}" ]; then
+  #bad rules.new directory
+  echo "The directory for node ${RUDDER_NODEID} new policies is empty"
+  exit 1;
+else 
+  case "${RUDDER_AGENT_TYPE}" in
+    "windows-dsc")
+      echo "compressing ${RUDDER_NEXT_POLICIES_DIRECTORY}" > /tmp/dschook
+      /usr/bin/zip -mr ${RUDDER_NEXT_POLICIES_DIRECTORY}/policies.zip ${RUDDER_NEXT_POLICIES_DIRECTORY}/*
```
Comment (ncharles) [inline rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies]: it should be rudder.zip

**Comment by ncharles** [comment]
Commit modified

File: rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Hooks parameter are passed by environment variable: 
+#
+# - RUDDER_GENERATION_DATETIME    : generation datetime: ISO-8601 YYYY-MM-ddTHH:mm:ss.sssZ date/time that identify that policy generation start 
+# - RUDDER_NODEID                 : the nodeId
+# - RUDDER_NEXT_POLICIES_DIRECTORY: new policies directory (for ex for nodes under root: /var/rudder/share/$RUDDER_NODEID/rules.new)
+# - RUDDER_AGENT_TYPE             : agent type ("cfengine-nova" or "cfengine-community")
+
+touch /tmp/compresshook
```
Comment (amousset) [inline rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies]: why?

Comment (ncharles) [inline rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies]: this is a debug that was not removed :(

**Comment by ncharles** [comment]
Commit modified

File: rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# - RUDDER_AGENT_TYPE             : agent type ("cfengine-nova" or "cfengine-community")
+
+if [ -z "${RUDDER_NEXT_POLICIES_DIRECTORY}" ]; then
+  #bad rules.new directory
+  echo "The directory for node ${RUDDER_NODEID} new policies is empty"
+  exit 1;
+else 
+  case "${RUDDER_AGENT_TYPE}" in
+    "windows-dsc")
+      echo "compressing ${RUDDER_NEXT_POLICIES_DIRECTORY}" > /tmp/dschook
```
Comment (amousset) [inline rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies]: same here I guess ;)

**Comment by ncharles** [comment]
Commit modified

File: rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies (hunk 3 above)
Comment (ncharles) [inline rudder-core/src/main/resources/hooks.d/policy-generation-node-ready/91-compress-dsc-policies]: indeed

**Pull request merged**